modeling/shared-global-algorithm.go

```diff
@@ -60,42 +60,34 @@ func (alg SharedGlobalAlgorithm) CreateSliceGroups(region regionInfo) (map[strin
 	// do so either.
 	var globalSliceGroup EndpointSliceGroup
 	globalSliceGroup.Label = "global"
-	globalSliceGroup.Composition = make(map[string]int)
-	globalSliceGroup.Weights = make(map[string]float64)
+	globalSliceGroup.Composition = make(map[string]weightedEndpoints)
+	globalSliceGroup.ZoneTrafficWeights = make(map[string]float64)
 	for name, zone := range region.zoneDetails {
-		var globalEndpoints int
+		var globalEndpoints weightedEndpoints
 		// If total pods <= threshold, all pods go to global slice
 		if region.totalEndpoints <= alg.globalThreshold {
-			globalEndpoints = zone.Endpoints
+			globalEndpoints.number = zone.Endpoints
+			globalEndpoints.weight = 1
 		} else {
 			// Otherwise calculate the global contribution of current zone based
 			// on the global weight and the deviation of this zone
-			globalEndpoints = int(math.Min(math.Max(0.0, deviation[name])/alg.globalWeight, float64(zone.Endpoints)))
+			globalEndpoints.number = int(math.Min(math.Max(0.0, deviation[name])/alg.globalWeight, float64(zone.Endpoints)))
```

Comment:
There's a lot of math going on here. Maybe add a comment to explain it and/or split it out into a couple lines.

Reply:
Added comments to explain